src/class-asset.php

```diff
@@ -36,13 +36,53 @@ public function enqueue_assets(): void {
 		$theme_prefix = $this->theme->get_prefix();
 
 		// Custom Fonts.
-		wp_enqueue_style( $theme_prefix . '-fonts-preload', $this->get_base_url() . '/dist/styles/fonts.css', false, $this->get_filemtime( 'styles/fonts.css' ) );
+		wp_enqueue_style( $theme_prefix . '-fonts-preload', $this->vite('assets/styles/fonts.css'), false);
 
 		// Theme Styles.
-		wp_enqueue_style( $theme_prefix . '-main-styles', $this->get_base_url() . '/dist/styles/main.css', array( $theme_prefix . '-fonts-preload' ), $this->get_filemtime( 'styles/main.css' ) );
+		wp_enqueue_style( $theme_prefix . '-main-styles', $this->vite( 'assets/styles/main.css' ), array( $theme_prefix . '-fonts-preload' ) );
 
 		// Theme Script.
-		wp_enqueue_script( $theme_prefix, $this->get_base_url() . '/dist/scripts/main.js', array(), $this->get_filemtime( 'scripts/main.js' ), true );
+		wp_enqueue_script( $theme_prefix, $this->vite( 'assets/scripts/main.ts' ), array(), null );
+	}
+
+	/**
+	 * Get the Vite processed file URL.
+	 *
+	 * @param string $file The file path.
+	 *
+	 * @return string The file URL.
+	 */
+	private function vite( string $file ): string {
+		if ( $this->is_vite_dev() ) {
+			if ( $file === 'assets/styles/fonts.css' ) {
+				return $this->get_base_url() . '/assets/styles/fonts.css';
+			}
+
+			if ( str_ends_with( $file, '.css' ) ) {
+				return 'https://localhost:5173/' . ltrim( $file, '/' );
+			}
+
+			return 'https://localhost:5173/' . ltrim( $file, '/' ) . '?import';
+		}
+
+		$manifest         = $this->get_base_path() . '/dist/.vite/manifest.json';
+		$manifest_content = json_decode( file_get_contents( $manifest ), true );
+
+		return $this->get_base_url() . '/dist/' . $manifest_content[ $file ]['file'];
+	}
```

Comment:
I think it would be nice to define a class constant to store the `https://localhost:5173/` value. I.e.

```php
/**
	 * Vite Dev Server URL.
	 *
	 * @var string
	 */
	protected const VITE_DEV_SERVER_URL = 'https://localhost:5173';
```

Then, this method could be refactored as
```suggestion
	private function vite( string $file ): string {
		if (! $this->is_vite_dev()) {
			$manifest         = $this->get_base_path() . '/dist/.vite/manifest.json';
			$manifest_content = json_decode( file_get_contents( $manifest ), true );

			return $this->get_base_url() . '/dist/' . $manifest_content[ $file ]['file'];
		}

		if ( $file === 'assets/styles/fonts.css' ) {
			return $this->get_base_url() . '/assets/styles/fonts.css';
		}

		if ( str_ends_with( $file, '.css' ) ) {
			return self::VITE_DEV_SERVER_URL . ltrim( $file, '/' );
		}

		return self::VITE_DEV_SERVER_URL . ltrim( $file, '/' ) . '?import';
	}
```